test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/InitializeTestProjectAttribute.cs

```diff
@@ -0,0 +1,103 @@
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Reflection;
+using Microsoft.AspNetCore.Testing;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
+{
+    public class InitializeTestProjectAttribute : BeforeAfterTestAttribute
+    {
+        private readonly string _projectName;
+
+        public InitializeTestProjectAttribute(string projectName)
+        {
+            _projectName = projectName;
+        }
+
+        public override void Before(MethodInfo methodUnderTest)
+        {
+            if (!typeof(MSBuildIntegrationTestBase).GetTypeInfo().IsAssignableFrom(methodUnderTest.DeclaringType.GetTypeInfo()))
+            {
+                throw new InvalidOperationException($"This should be used on a class derived from {typeof(MSBuildIntegrationTestBase)}");
+            }
+
+            var solutionRoot = TestPathUtilities.GetSolutionRootDirectory("Razor");
+            if (solutionRoot == null)
+            {
+                throw new InvalidOperationException("Could not find solution root.");
+            }
+
+            var projectRoot = Path.Combine(solutionRoot, "test", "testapps", _projectName);
+            if (!Directory.Exists(projectRoot))
+            {
+                throw new InvalidOperationException($"Could not find project at '{projectRoot}'");
+            }
+
+            var tempDirectory = ProjectDirectory.Create();
+
+            try
+            {
+                CopyDirectory(new DirectoryInfo(projectRoot), new DirectoryInfo(tempDirectory.FilePath));
+
+                foreach (var project in Directory.EnumerateFiles(tempDirectory.FilePath, "*.csproj"))
+                {
+                    RewriteCsproj(project);
+                }
+
+                MSBuildIntegrationTestBase.Project = tempDirectory;
+            }
+            catch
+            {
+                tempDirectory.Dispose();
+                throw;
+            }
+
+            void CopyDirectory(DirectoryInfo source, DirectoryInfo destination)
+            {
+                foreach (var directory in source.EnumerateDirectories())
+                {
+                    if (directory.Name == "bin" || directory.Name == "obj")
+                    {
+                        // Just in case someone has opened the project in an IDE or built it. We don't want to copy
+                        // these folders.
+                        continue;
+                    }
+
+                    var created = destination.CreateSubdirectory(directory.Name);
+                    CopyDirectory(directory, created);
+                }
+
+                foreach (var file in source.EnumerateFiles())
+                {
+                    file.CopyTo(Path.Combine(destination.FullName, file.Name));
+                }
+            }
+
+            void RewriteCsproj(string filePath)
```

Comment:
ew. Just create a `Directory.Build.props` (gets automatically imported) that sets this property in the base `ProjectDirectory`

Reply:
That doesn't work. You're **moving** the project

Reply:
Create the Directory.Build.props to the location that you moved the project. Maybe I'm missing something.